### PR TITLE
Ticket #22990 sensitive post

### PR DESCRIPTION
**Pull request by ziima**

### CODE REVIEW

File: django/http/request.py [hunk 1]
```diff
@@ -468,15 +468,22 @@ def build_request_repr(request, path_override=None, GET_override=None,
...
     else:
         try:
-            post = (pformat(POST_override)
-                    if POST_override is not None
-                    else pformat(request.POST))
+            if POST_override is not None:
+                post = pformat(POST_override)
+            else:
+                # Use the exception reporter filter to return the cleansed POST data
+                from django.views.debug import get_exception_reporter_filter
```
Comment (berkerpeksag) [inline django/http/request.py]: If there is no danger of a circular import error, I'd move this to the top of the file.

File: tests/view_tests/tests/test_debug.py [hunk 2]
```diff
@@ -596,72 +580,89 @@ class ExceptionReporterFilterTests(TestCase, ExceptionReportTestMixin):
     Ensure that sensitive information can be filtered out of error reports.
     Refs #14614.
     """
-    rf = RequestFactory()
+    client_class = SensitiveTestClient
+
+    def setUp(self):
+        logger = logging.getLogger('django.request')
+        log_handler = logger.handlers[0]
```
Comment (berkerpeksag) [inline tests/view_tests/tests/test_debug.py]: I've backported `assertLogs` in #3467. Perhaps we can merge `assertLogs` part of the PR first.

File: tests/view_tests/tests/test_debug.py [hunk 3]
```diff
@@ -596,72 +580,89 @@ class ExceptionReporterFilterTests(TestCase, ExceptionReportTestMixin):
...
+        # Disable filter for DEBUG=True
+        filters = log_handler.filters
+        self.addCleanup(lambda: setattr(log_handler, 'filters', filters))
```
Comment (berkerpeksag) [inline tests/view_tests/tests/test_debug.py]: I'd move this monkeypatch stuff to `tearDown`. `addCleanup` is suitable for directory, file or another resource cleanups.

File: tests/view_tests/tests/test_debug.py [hunk 4]
```diff
@@ -703,13 +706,30 @@ def test_sensitive_function_keyword_arguments(self):
...
+            self.verify_unsafe_email('/sensitive/sensitive_kwargs/')
 
         with self.settings(DEBUG=False):
-            self.verify_safe_response(sensitive_kwargs_function_caller, check_for_POST_params=False)
-            self.verify_safe_email(sensitive_kwargs_function_caller, check_for_POST_params=False)
+            self.verify_public_response('/sensitive/sensitive_kwargs/', text)
+            self.verify_safe_email('/sensitive/sensitive_kwargs/',
+                                   check_for_POST_params=False)
+
+    def test_sensitive_class_view(self):
```
Comment (berkerpeksag) [inline tests/view_tests/tests/test_debug.py]: I'd split the test into two tests.

**Comment by berkerpeksag** [comment]
General comments:

* Could you squash the commits into one commit? You also need to fix merge conflict.
* Needs a release note
* Commit message format should follow `Fixed #XXXX --  ...`. Please read it at https://docs.djangoproject.com/en/dev/internals/contributing/committing-code/#committing-guidelines

**Comment by timgraham** [comment]
Closing in absence of follow up. Please send a new PR if you someone can update this, thanks!

**Pull request closed without merging**